packages/block-editor/src/components/list-view/block-select-button.js

```diff
@@ -72,14 +77,23 @@ function ListViewBlockSelectButton(
 			>
 				<ListViewExpander onClick={ onToggleExpanded } />
 				<BlockIcon icon={ blockInformation?.icon } showColors />
-				<span className="block-editor-list-view-block-select-button__title">
-					<BlockTitle clientId={ clientId } maximumLength={ 35 } />
-				</span>
-				{ blockInformation?.anchor && (
-					<span className="block-editor-list-view-block-select-button__anchor">
-						{ blockInformation.anchor }
-					</span>
-				) }
+				<HStack
+					alignment="center"
+					justify="flex-start"
+					className="block-editor-list-view-block-select-button__label-wrapper"
+				>
+					<Truncate
```

Comment:
I've switched this to use `Truncate` instead of `BlockTitle` — in this context, due to the variable width of the available space, we want to use the `auto` ellipsizeMode instead of truncating to a particular number of characters, so this way gives us a bit more flexibility.

Reply:
I see I've missed a couple of tests that also use XPaths expecting a particular sibling of the label. E.g.

* https://github.com/WordPress/gutenberg/blob/b07ea2cf7d17b4f12855fce8d9baa36d86c803a6/packages/e2e-tests/specs/editor/blocks/navigation.test.js#L1054 and
* https://github.com/WordPress/gutenberg/blob/a92b73ceb9e58dc600fb16f2865b2716ee67cef0/packages/e2e-tests/specs/editor/blocks/gallery.test.js#L106

I'll fix them up tomorrow